src/api/app/controllers/webui/users/announcements_controller.rb

```diff
@@ -0,0 +1,8 @@
+class Webui::Users::AnnouncementsController < Webui::WebuiController
+  before_action :require_login
+
+  def create
+    announcement = Announcement.find(params[:id])
```

Comment:
This will throw an exception (and render our 404.html) if there is no such announcement.